### ISSUE TITLE
Strikethrough not work with iTerm2

### ISSUE THREAD

**Issue by chemzqm** [neovim/neovim#14550]
- `nvim --version`: v0.5.0-dev+1312-gd2be261e8
- Operating system/version: MacOS 10.15
- Terminal name/version: iTerm2
- `$TERM`: xterm-256color

[x] `vim -u DEFAULTS` (version: 8.1.1719) behaves differently

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC
call setline(1, 'abc')
hi CocStrike  guifg=#989898 ctermfg=gray  cterm=strikethrough gui=strikethrough
call matchaddpos('CocStrike', [1])
```

### Actual behaviour
<img width="57" alt="Screen Shot 2021-05-14 at 10 09 10 AM" src="https://user-images.githubusercontent.com/251450/118210057-6fe35780-b49c-11eb-9536-f628e0f29395.png">

### Expected behaviour
<img width="51" alt="Screen Shot 2021-05-14 at 10 08 45 AM" src="https://user-images.githubusercontent.com/251450/118210060-72de4800-b49c-11eb-805c-ec567d6eacd7.png">

Using vim8 or neovim on linux doesn't have this issue.

**Comment by mhinz** [comment]
You probably have `t_Ts`/`t_Te` set for Vim?

Your given example works here (inside and outside of tmux) with iTerm2, so I'm assuming that your `$TERM` is missing the correct entries for the strikethrough mode.

```
$ infocmp -1x | egrep '(sm|rm)xx'
	rmxx=\E[29m,
	smxx=\E[9m,
```

If the above command gives you no output, strikethrough mode won't work. E.g. `tput smxx; echo foo; tput rmxx` should give you a struck "foo".

**Comment by mhinz** [comment]
Since this is a configuration issue, I'm closing this.

You can use [this script](https://github.com/mhinz/dotfiles/blob/master/bin/fix-term) (just run it without arguments). It takes the current entries for `xterm-256color` and `tmux-256color`, adds a few capabilities for italics and strikethrough (see the comment at the top) and then creates new terminfo files:

```
$ tree ~/.terminfo
/Users/mhi/.terminfo
├── 74
│   └── tmux-256color
└── 78
    └── xterm-256color
```

So, when using `TERM=xterm-256color` in iTerm2, it should use `~/.terminfo/78/tmux-256color` automatically. You can double-check that with:

```
$ infocmp | head -n1
#	Reconstructed via infocmp from file: /Users/mhi/.terminfo/78/xterm-256color
```

**Comment by carbon-steel** [comment]
I got the populated results from running `infocmp`:
```
$ infocmp -1x | egrep '(sm|rm)xx'
	rmxx=\E[29m,
	smxx=\E[9m,
```
which is supposed to mean that strikethrough should work. However, it wasn't working. I tried running [your script](https://github.com/mhinz/dotfiles/blob/master/bin/fix-term) anyway and IT WORKED! I've spend hours today and yesterday trying to figure this out. Thank you so much!

It is now on my todo list to learn more about terminfo :)
